extensions/pkg/webhook/utils.go

```diff
@@ -63,80 +65,93 @@ func SerializeCommandLine(command []string, n int, sep string) string {
 	return strings.Join(command[0:n], " ") + " " + strings.Join(command[n:], sep)
 }
 
-// ContainerWithName returns the container with the given name if it exists in the given slice, nil otherwise.
+// ContainerWithName returns the first occurrence of a container with the given name if it exists in the given slice, nil otherwise.
 func ContainerWithName(containers []corev1.Container, name string) *corev1.Container {
-	if i := containerWithNameIndex(containers, name); i >= 0 {
-		return &containers[i]
+	for i, container := range containers {
+		if container.Name == name {
+			return &containers[i]
+		}
 	}
 	return nil
 }
 
-// PVCWithName returns the PersistentVolumeClaim with the given name if it exists in the given slice, nil otherwise.
+// PVCWithName returns the first occurrence of a PersistentVolumeClaim with the given name if it exists in the given slice, nil otherwise.
```

Comment:
```suggestion
// PVCWithName returns the first PersistentVolumeClaim with the specified name from the slice, or nil if not found.
```
Similar changes at other places also.